client/packages/system/src/Patient/Encounter/DetailModal.tsx

```diff
@@ -1,95 +1,144 @@
 import { ProgramRowFragmentWithId, usePatient } from '../api';
-import { SaveDocumentMutation, useJsonForms } from '../JsonForms';
+import { CreateDocument, useJsonForms } from '../JsonForms';
 import { ProgramSearchInput } from '../Components';
 
-const useUpsertEncounter = (
-  patientId: string,
-  programType: string,
-  type: string
-): SaveDocumentMutation => {
-  const { mutateAsync: insertEncounter } = useEncounter.document.insert();
-  const { mutateAsync: updateEncounter } = useEncounter.document.update();
-  return async (jsonData: unknown, formSchemaId: string, parent?: string) => {
-    if (parent === undefined) {
-      const result = await insertEncounter({
-        data: jsonData,
-        schemaId: formSchemaId,
-        patientId,
-        type,
-        programType,
-      });
-      return result;
-    } else {
-      const result = await updateEncounter({
-        data: jsonData,
-        parent,
-        schemaId: formSchemaId,
-      });
-      return result;
-    }
-  };
-};
-
 export const EncounterDetailModal: FC = () => {
   const patientId = usePatient.utils.id();
-
-  const { current, document, programType, reset, setProgramType } =
-    usePatientModalStore();
-
+  const t = useTranslation('patients');
+  const { current, reset, document } = usePatientModalStore();
   const [program, setProgram] = useState<ProgramRowFragmentWithId | null>(null);
-  const handleSave = useUpsertEncounter(
+  const [error, setError] = useState(false);
+  const { mutate: fetch, isLoading, data } = useEncounter.registry.byProgram();
+  const isCreate = !document?.name;
+  const [createDoc, setCreateDoc] = useState<CreateDocument | undefined>(
+    undefined
+  );
+
+  const handleSave = useEncounter.document.upsert(
     patientId,
-    programType ?? '',
-    document?.type ?? ''
+    program?.type ?? '',
+    createDoc?.documentRegistry?.documentType ?? ''
+  );
+  const { JsonForm, saveData, isDirty, validationError } = useJsonForms(
+    document?.name,
+    {
+      handleSave,
+    },
+    createDoc
   );
-  const { JsonForm, saveData, isLoading, isDirty, validationError } =
-    useJsonForms(
-      document?.name,
-      {
-        handleSave,
-      },
-      document?.createDocument
-    );
+
+  const onClose = () => {
+    reset();
+    setProgram(null);
+    setCreateDoc(undefined);
+    setError(false);
+  };
 
   const { Modal } = useDialog({
     isOpen: current === PatientModal.Encounter,
-    onClose: reset,
+    onClose,
   });
 
   const onChangeProgram = (program: ProgramRowFragmentWithId) => {
     setProgram(program);
-    setProgramType(program.type);
+    setError(false);
+    fetch(program.document.documentRegistry?.id || '');
   };
 
+  useEffect(() => {
+    if (data && data?.totalCount > 0) {
+      const documentRegistry = data.nodes[0];
+      if (documentRegistry) {
+        setCreateDoc({ data: {}, documentRegistry });
+        setError(false);
+        return;
+      }
+    } else {
+      setError(true);
```

Comment:
the `placeholder program` encounter doesn't have a doc registry entry so I'm showing an error